### PR TITLE
Feat/security update august 2023

### PR DESCRIPTION
**Pull request by pavlovcik**
Related https://github.com/ubiquity/ubiquibot/pull/622

Cosmetic hotfix for 622 should be merged in next: https://github.com/ubiquity/ubiquibot/pull/636

I had a merge conflict with `src/adapters/supabase/helpers/client.ts` around line 170-180. I accepted all incoming but I hope it didn't break anything. This should be double checked. 

I'm extremely tired but its urgent and here's some context that might be helpful from some diffs.

image

### CODE REVIEW

File: src/adapters/supabase/helpers/client.ts [hunk 1]
```diff
@@ -175,7 +175,7 @@ export const upsertWalletAddress = async (username: string, address: string): Pr
       created_at: new Date().toUTCString(),
       updated_at: new Date().toUTCString(),
     });
-    logger.info(`Creating a new wallet_table record done, { data: ${_data}, error: ${_error} }`);
+    logger.info(`Creating a new wallet_table record done, { error: ${_error?.message} }`);
```
Comment (pavlovcik) [inline src/adapters/supabase/helpers/client.ts:178]: I'm assuming the issue is here. 
```suggestion
    logger.info(`Creating a new wallet_table record done, { data: ${_data}, error: ${_error?.message} }`);
```

File: src/handlers/pricing/action.ts [hunk 2]
```diff
@@ -46,3 +57,8 @@ export const pricingLabelLogic = async (): Promise<void> => {
     logger.info(`Skipping action...`);
   }
 };
+
+export const isParentIssue = (body: string) => {
+  const parentPattern = /-\s+\[( |x)\]\s+#\d+/;
```
Comment (pavlovcik) [inline src/handlers/pricing/action.ts:62]: This is just checking for a checklist? This should look for the UI element when GitHub shows that it is a "tracking issue" of other issues?

File: src/helpers/issue.ts [hunk 3]
```diff
@@ -314,6 +314,49 @@ export const removeAssignees = async (issue_number: number, assignees: string[])
...
+  try {
+    const res = await context.octokit.rest.repos.checkCollaborator({
+      owner: payload.repository.owner.login,
+      repo: payload.repository.name,
+      username,
+    });
+
+    return res.status === 204;
+  } catch (e: unknown) {
+    logger.error(`Checking if user permisson for repo failed!, reason: ${e}`);
```
Comment (pavlovcik) [inline src/helpers/issue.ts:337]: - I handled all spelling errors in #643

**Comment by pavlovcik** [comment]
Because this is urgent, and because the commits already passed pull request reviews, I'll merge now. If a fatal issue is found please do not hesitate to revert this @0xcodercrane

File: src/handlers/pricing/action.ts (hunk 2 above)
Comment (wannacfuture) [reply src/handlers/pricing/action.ts:62]: Yeah, usually parent issue conatins this string:
`- [ ] #issue_number`.

Comment (pavlovcik) [reply src/handlers/pricing/action.ts:62]: Looks like this achieved the specification. I was concerned about this UI element but perhaps it isn't necessary to implement. 
- https://github.com/ubiquity/ubiquibot/issues/268#issuecomment-1585532148

**Comment by 0xcodercrane** [comment]
> Because this is urgent, and because the commits already passed pull request reviews, I'll merge now. If a fatal issue is found please do not hesitate to revert this @0xcodercrane
> 
> Retroactive review required

I will have a couple of action items against production bot in ubiquibot/production repo.

**Comment by 0xcodercrane** [comment]
These changes have been merged into the development branch as well? or @pavlovcik did a cherry-pick?

**Comment by pavlovcik** [comment]
I basically just merged from `development` to `main` 

But I technically did it from my fork. So I branched from `development` into `feat/security-update...` and then merged that into this `main` branch.

**Comment by 0xcodercrane** [comment]
makes sense. you're right.

**Pull request merged**